lib/screenplay/config/local_fetch.ex

```diff
@@ -116,4 +116,30 @@ defmodule Screenplay.Config.LocalFetch do
       {:error, _} -> :error
     end
   end
+
+  defp add_labels_to_config(config) do
+    with {:ok, label_contents, _} <- do_get(:local_paess_labels_file_spec),
+         {:ok, labels} <- do_decode(label_contents, :local_paess_labels_file_spec) do
+      Enum.reduce(
+        config,
+        [],
+        fn place, acc ->
+          new_screens =
+            Enum.reduce(
+              place["screens"],
+              [],
+              fn screen, acc ->
+                [Map.put(screen, "label", Map.get(labels, screen["id"])) | acc]
+              end
+            )
+
+          [Map.put(place, "screens", new_screens) | acc]
+        end
+      )
```

Comment:
**question:** Do we need `reduce` here? These look like straight-forward `Enum.map`s to me (with the exception that they also reverse the lists processed but I imagine that's just a side effect of using `reduce` in the first place).

```elixir
Enum.map(config, fn place ->
  new_screens =
    Enum.map(place["screens"], fn screen ->
      Map.put(screen, "label", Map.get(labels, screen["id"]))
    end)

  Map.put(place, "screens", new_screens)
end)
```

Additionally, this might be simplified further with [`update_in/2`](https://hexdocs.pm/elixir/Kernel.html#update_in/2) (or [`update_in/3`](https://hexdocs.pm/elixir/Kernel.html#put_in/3))

Reply:
No idea why I've been so `reduce`-happy lately. Will work on changing it.